assets/stylesheets/layout/_main.scss

```diff
@@ -372,15 +381,15 @@ $sidebar-width-min: 228px;
 .wp-secondary .site-selector {
 	background: lighten( $gray, 30% );
 	border-right: 1px solid lighten( $gray, 25% );
-	position: fixed;
-		top: 47px;
+	position: absolute;
```

Comment:
Ok, the biggest change here is that the sidebar now knows how much width to take base on the flexbox it's contained inside.

Mobile Safari, OS X Chrome, OS X Firefox, OS X Safari are all working the same as before the flexbox change.